controller/src/main/java/io/pravega/controller/store/stream/ZKStream.java

```diff
@@ -422,12 +438,69 @@ public void refresh() {
 
     @Override
     CompletableFuture<Void> createCompletedTxEntry(final UUID txId, final TxnStatus complete, final long timestamp) {
-        final String completedTxPath = getCompletedTxPath(txId.toString());
-        return store.createZNodeIfNotExist(completedTxPath,
-                new CompletedTxnRecord(timestamp, complete).toByteArray())
-                .whenComplete((r, e) -> cache.invalidateCache(completedTxPath));
+        String root = String.format(STREAM_COMPLETED_TX_BATCH_PATH, currentBatchSupplier.get(), getScope(), getName());
+        String path = ZKPaths.makePath(root, txId.toString());
+
+        CompletableFuture<Void> future1 = store.createZNodeIfNotExist(path,
+                        new CompletedTxnRecord(timestamp, complete).toByteArray())
+                        .whenComplete((r, e) -> cache.invalidateCache(completedTxPathOldScheme));
+
+        // region Backward Compatibility
+        // TODO 2755 retire code in this region 
+        // We will continue to write to old path for backward compatibility.
+        // This should eventually be removed.
+        if (!Config.DISABLE_COMPLETED_TXN_BACKWARD_COMPATIBILITY) {
+            final String completedTxPath = getOldSchemeCompletedTxPath(txId.toString());
+
+            CompletableFuture<Void> future2 = store.createZNodeIfNotExist(completedTxPath,
```

Comment:
Given that this is the future that is supposed to stay long term, we can name it `createZnodeFuture` or just `createFuture`.